admin-dev/themes/new-theme/js/components/grid/extension/action/row/submit-row-action-extension.js

```diff
@@ -39,29 +41,62 @@ export default class SubmitRowActionExtension {
       event.preventDefault();
 
       const $button = $(event.currentTarget);
-      const confirmMessage = $button.data('confirm-message');
-
-      if (confirmMessage.length && !window.confirm(confirmMessage)) {
-        return;
-      }
+      const confirmMessage = $button.data('confirmMessage');
+      const confirmTitle = $button.data('title');
 
       const method = $button.data('method');
       const isGetOrPostMethod = ['GET', 'POST'].includes(method);
+      if (confirmTitle) {
+        this.showConfirmModal($button, grid, confirmMessage, confirmTitle, isGetOrPostMethod, method);
+      } else {
+        if (confirmMessage.length && !window.confirm(confirmMessage)) {
+          return;
+        }
 
-      const $form = $('<form>', {
-        action: $button.data('url'),
-        method: isGetOrPostMethod ? method : 'POST',
-      }).appendTo('body');
-
-      if (!isGetOrPostMethod) {
-        $form.append($('<input>', {
-          type: '_hidden',
-          name: '_method',
-          value: method,
-        }));
+        this.postForm($button, isGetOrPostMethod, method);
       }
-
-      $form.submit();
     });
   }
+
+  postForm($button, isGetOrPostMethod, method) {
```

Comment:
```suggestion
  postForm($button, method) {
```

Reply:
You don't need to pass a boolean checking if it's a GET or POST, when you already have the `method` variable, can tells you if it's a GET or POST. :thinking: